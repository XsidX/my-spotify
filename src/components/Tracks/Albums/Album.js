import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { solid } from '@fortawesome/fontawesome-svg-core/import.macro';
import propTypes from 'prop-types';

const Album = ({ album }) => {
  const { name, images, release_date: dateReleased, type } = album;
  // trim the date to only show the year
  const yearReleased = dateReleased.substring(0, 4);
  return (
    <div className="text-white flex items-center gap-3 drop-shadow-md relative z-30">
      <div className="h-24 w-20">
        <img src={images[0].url} alt={name} />
      </div>
      <div className="text-sm">
        <p className="">{name}</p>
        <div className="text-gray-300 text-xs flex items-center gap-1">
          <p>{yearReleased}</p>
          <FontAwesomeIcon icon={solid('circle')} className="text-tiny" />
          <p className="capitalize">{type}</p>
        </div>
      </div>
    </div>
  );
};
Album.propTypes = {
  album: propTypes.shape({
    name: propTypes.string.isRequired,
    images: propTypes.arrayOf(
      propTypes.shape({
        url: propTypes.string.isRequired,
      }),
    ).isRequired,
    release_date: propTypes.string.isRequired,
    type: propTypes.string.isRequired,
  }).isRequired,
};

export default Album;
