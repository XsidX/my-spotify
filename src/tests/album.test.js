import renderer from 'react-test-renderer';
import Album from '../components/Tracks/Albums/Album';

describe('Album', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Album
          album={{
            name: 'album name',
            images: [{ url: 'url' }],
            release_date: '2021-01-01',
            type: 'album',
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
