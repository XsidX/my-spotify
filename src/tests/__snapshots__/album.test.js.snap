// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album renders correctly 1`] = `
<div
  className="text-white flex items-center gap-3 drop-shadow-md relative z-30"
>
  <div
    className="h-24 w-20"
  >
    <img
      alt="album name"
      src="url"
    />
  </div>
  <div
    className="text-sm"
  >
    <p
      className=""
    >
      album name
    </p>
    <div
      className="text-gray-300 text-xs flex items-center gap-1"
    >
      <p>
        2021
      </p>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle text-tiny"
        data-icon="circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <p
        className="capitalize"
      >
        album
      </p>
    </div>
  </div>
</div>
`;
